src/gsi/m_berror_stats_reg.f90

```diff
@@ -807,103 +819,126 @@ subroutine berror_read_wgt_reg(msig,mlat,corz,corp,hwll,hwllp,vz,rlsig,varq,qopt
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwll(i,1,nrf3_q)
+           hwllp(i,loc)=hwll(i,1,nrf3_q)
         end do
-     endif
 
-     if (cvarsmd(n)=='gustwter') then
+     else if (cvarsmd(n)=='gustwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_gust)
+           corp(i,loc)=corp(i,nrf2_gust)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_gust)
+           hwllp(i,loc)=hwllp(i,nrf2_gust)
         end do
-     endif
 
-     if (cvarsmd(n)=='wspd10mwter') then
+     else if (cvarsmd(n)=='wspd10mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_wspd10m)
+           corp(i,loc)=corp(i,nrf2_wspd10m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_wspd10m)
+           hwllp(i,loc)=hwllp(i,nrf2_wspd10m)
         end do
-     endif
 
-     if (cvarsmd(n)=='td2mwter') then
+     else if (cvarsmd(n)=='td2mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_td2m)
+           corp(i,loc)=corp(i,nrf2_td2m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_td2m)
+           hwllp(i,loc)=hwllp(i,nrf2_td2m)
         end do
-     endif
 
-     if (cvarsmd(n)=='mxtmwter') then
+     else if (cvarsmd(n)=='mxtmwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_mxtm)
+           corp(i,loc)=corp(i,nrf2_mxtm)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_mxtm)
+           hwllp(i,loc)=hwllp(i,nrf2_mxtm)
         end do
-     endif
 
-     if (cvarsmd(n)=='mitmwter') then
+     else if (cvarsmd(n)=='mitmwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_mitm)
+           corp(i,loc)=corp(i,nrf2_mitm)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_mitm)
+           hwllp(i,loc)=hwllp(i,nrf2_mitm)
         end do
-     endif
 
-     if (cvarsmd(n)=='uwnd10mwter') then
+     else if (cvarsmd(n)=='uwnd10mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_uwnd10m)
+           corp(i,loc)=corp(i,nrf2_uwnd10m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_uwnd10m)
+           hwllp(i,loc)=hwllp(i,nrf2_uwnd10m)
         end do
-     endif
 
-     if (cvarsmd(n)=='vwnd10mwter') then
+     else if (cvarsmd(n)=='vwnd10mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_vwnd10m)
+           corp(i,loc)=corp(i,nrf2_vwnd10m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_vwnd10m)
+           hwllp(i,loc)=hwllp(i,nrf2_vwnd10m)
         end do
-     endif
+!  if not found use default
+     else 
+        do i=1,mlat
+           corp(i,loc)=corz_default
+        end do
+        do i=0,mlat+1
+           hwllp(i,loc)=hwll_default
+        end do
+     end if
   enddo
 
   deallocate(nrf3_loc,nrf2_loc,nmotl_loc)
+! Normalize vz with del sigmma and convert to vertical grid units!
+  if( .not. usenewgfsberror)then
```

Comment:
what is the case for usenewgfsberror=.true. ?  I couldn't find this normalization of vz for it.

Reply:
You are correct, there is no normalization for vz when the new gfs berror file is used.  I could find no such normalization in the global runs and this result produced reasonable results.  If one is necessary, it can be added, but this whole code makes little sense to me.

Reply:
John, Thanks!  Did you mean this normalization is not necessary even for the "old" bkerr files ?  From the perspective of coding,  would it be preferable to have the same treatment to both the "old" and "new" bkerr files if there are no changes for the definitions of vz and related variables?

Reply:
I did not mean that it is not necessary for the old bkerr files.  In fact, as I think about it more, it probably should be there for the new one as well.  I could remove the check on usenewgffsberror = .true..  But with this change, the single obs plots should be produced to make sure that this produces reasonable results.  I cannot produce these plots.  Maybe we can have Xiaoyan produce these plots to make sure this change results in everything being OK.

Reply:
John, Thanks.  Your clarification does make sense to me!